tests/integration/apigateway/test_apigateway_integrations.py

```diff
@@ -652,6 +664,191 @@ def test_put_integration_validation(apigateway_client):
+    def _check_available():
+        result = ec2_client.describe_vpc_endpoints(VpcEndpointIds=[endpoint_id])
+        endpoint_details = result["VpcEndpoints"][0]
+        # may have multiple entries in AWS
+        endpoint_details["DnsEntries"] = endpoint_details["DnsEntries"][:1]
+        endpoint_details.pop("SubnetIds", None)
+        endpoint_details.pop("NetworkInterfaceIds", None)
+        assert endpoint_details["State"] == "available"
+        snapshot.match("endpoint-details", endpoint_details)
+
+    retry(_check_available, retries=30, sleep=poll_sleep)
+
+    # update API with VPC endpoint
+    patches = [
+        {"op": "replace", "path": "/endpointConfiguration/types/EDGE", "value": "PRIVATE"},
+        {"op": "add", "path": "/endpointConfiguration/vpcEndpointIds", "value": endpoint_id},
+    ]
+    apigateway_client.update_rest_api(restApiId=api_id, patchOperations=patches)
+
+    # create Lambda that invokes API via VPC endpoint (required as the endpoint is only accessible within the VPC)
+    subdomain = f"{api_id}-{endpoint_id}"
+    endpoint = api_invoke_url(subdomain, stage=DEFAULT_STAGE_NAME, path="/test")
+    host_header = urlparse(endpoint).netloc
+
+    # create Lambda function that invokes the API GW (private VPC endpoint not accessible from outside of AWS)
+    if not is_aws_cloud():
+        if config.LAMBDA_EXECUTOR == "local" and is_old_provider():
+            # special case: return localhost for local Lambda executor (TODO remove after full switch to v2 provider)
+            api_host = LOCALHOST
+        else:
+            api_host = get_main_endpoint_from_container()
+        endpoint = endpoint.replace(host_header, f"{api_host}:{config.get_edge_port_http()}")
+    lambda_code = textwrap.dedent(
+        f"""
+    def handler(event, context):
+        import requests
+        headers = {{"content-type": "application/json", "host": "{host_header}"}}
+        result = requests.post("{endpoint}", headers=headers)
+        return {{"content": result.content.decode("utf-8"), "code": result.status_code}}
+    """
+    )
+    func_name = f"test-{short_uid()}"
+    vpc_config = {
+        "SubnetIds": subnets,
+        "SecurityGroupIds": [security_group],
+    }
+    create_lambda_function(
+        func_name=func_name, handler_file=lambda_code, timeout=10, VpcConfig=vpc_config
+    )
+
+    # create resource policy
+    statement = {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Effect": "Allow",
+                "Principal": "*",
+                "Action": "execute-api:Invoke",
+                "Resource": ["execute-api:/*"],
+            }
+        ],
+    }
+    patches = [{"op": "replace", "path": "/policy", "value": json.dumps(statement)}]
+    result = apigateway_client.update_rest_api(restApiId=api_id, patchOperations=patches)
+    result["policy"] = json.loads(to_bytes(result["policy"]).decode("unicode_escape"))
+    snapshot.match("api-details", result)
```

Comment:
nit: could we add this jsonpatch operation to the previous call to `update_rest_api`? Honestly, this does not really matter to me, was just a thought. 😄

Reply:
Yep, could totally be added above as well. 👍 Just separated it out to group the setup into logical units..

Reply:
Makes sense 👍